cernopendata_client/cli.py

```diff
@@ -26,14 +26,19 @@
 
 SEARCH_URL = 'http://opendata.cern.ch/api/records/'
 
-
+# Callback
 def ensure_positive_int(ctx, param, value):
-    if value < 0:
-        click.secho("Recid should be a positive integer", fg='red', err=True)
-        sys.exit()
-    else:
+        if value < 0:
+            click.secho("Recid should be a positive integer", fg='red', err=True)
+            sys.exit()
         return value
 
+# Custom Type
+class ArgumentCheck(click.ParamType):
+    def ensure_positive_int(self, value, param, ctx):
+        if value < 0:
```

Comment:
Please move "Custom Type" comment into a class docstring explaining what the class does.